### PR TITLE
[Fix variable process] Allow to free a dof

### PR DESCRIPTION
**Pull request by miguelmaso**
**Description**
Running an example I realized that I need to free a dof in a corner. Is it possible to do that modification to adapt the existing process?
The default behavior does not change.

**Changelog**
- Allow to free a scalar dof if specified
- Corresponding changes in the vector process to be consistent

### CODE REVIEW

File: kratos/python_scripts/fix_scalar_variable_process.py [hunk 1]
```diff
@@ -52,6 +53,9 @@ def ExecuteInitializeSolutionStep(self):
             is_fixed = self.settings["constrained"].GetBool()
             if is_fixed:
                 self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
+            else:
+                if self.settings["free_dof_if_specified"].GetBool():
+                    self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
```
Comment (philbucher) [inline kratos/python_scripts/fix_scalar_variable_process.py:58]:
```suggestion
            self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
```

wouldn't this do the same?

Comment (miguelmaso) [reply kratos/python_scripts/fix_scalar_variable_process.py:58]: yes, but it would change the previous behavior. That's what I did locally

**Comment by RiccardoRossi** [comment]
@miguelmaso currently the processes leave the dofs in free state at the end of every step.

so in principle you can have something free by not fixing it in the step you want.

i would prefer not to touch the general processes that are used by everyone, so let's see first if there is any other way out

File: kratos/python_scripts/fix_scalar_variable_process.py (hunk 1 above)
Comment (miguelmaso) [reply kratos/python_scripts/fix_scalar_variable_process.py:58]: what is the expected behavior if `contrained` is set to `false`?

**Review by RiccardoRossi — CHANGES_REQUESTED** [review]
for example instead of specifying the dof not to be fixed, why don't you provide a submodelpart which only contains the dofs that you actually need to fix?

**Comment by miguelmaso** [comment]
I want to exclude a corner, which is the node that defines a boundary condition (the ending node of a line). I find easier to free the corner than excluding the corner from the sub model part

File: kratos/python_scripts/fix_scalar_variable_process.py (hunk 1 above)
Comment (RiccardoRossi) [reply kratos/python_scripts/fix_scalar_variable_process.py:58]: it will not fix it

**Comment by miguelmaso** [comment]
Another possibility would be to add another process, FreeScalarVariableProcess, e.g.

**Comment by RiccardoRossi** [comment]
If you had a process that just does the freeing you will have to take care about the order at which you are calling them

I would be more in favor of an extended process which does both the fixing and the freeing (which can use internally the one to fix dofs, and have the interface you propose)

**Comment by miguelmaso** [comment]
Just now I am thinking in a process to do operations with the submodelparts (union, intersection, difference) that will be executed before the BC processes. I think it is more general and aligned with your comment:

> for example instead of specifying the dof not to be fixed, why don't you provide a submodelpart which only contains the dofs that you actually need to fix?

**Comment by miguelmaso** [comment]
> Just now I am thinking in a process to do operations with the submodelparts (union, intersection, difference) that will be executed before the BC processes. I think it is more general and aligned with your comment:

it is in #8608

**Pull request closed without merging**